app/src/main/java/tech/relaycorp/gateway/domain/publicsync/CollectParcelsFromGateway.kt

```diff
@@ -37,17 +41,31 @@ class CollectParcelsFromGateway
             poWebClient.use {
                 poWebClient
                     .collectParcels(arrayOf(signer), streamingMode)
+                    .retry(Long.MAX_VALUE) { e ->
+                        if (keepAlive && e is ServerException) {
+                            logger.log(
+                                Level.WARNING,
+                                "Could not collect parcels due to server error, will retry.",
+                                e
+                            )
+                            delay(RETRY_AFTER_PERIOD)
+                            true
+                        } else false
+                    }
                     .collect { collectParcel(it, keepAlive) }
             }
         } catch (e: ServerException) {
-            logger.log(Level.INFO, "Could not collect parcels due to server error", e)
+            logger.log(Level.SEVERE, "Could not collect parcels due to server error", e)
             return
         } catch (e: ClientBindingException) {
             logger.log(Level.SEVERE, "Could not collect parcels due to client error", e)
             return
         } catch (e: NonceSignerException) {
             logger.log(Level.SEVERE, "Could not collect parcels due to signing error", e)
             return
+        } catch (e: Exception) {
```

Comment:
I'd suggest to use `PDCException` instead, so that we limit this to PoWeb exceptions without silently masking app bugs.